ModuleNotFoundError: No module named 'ansible.module_utils.testing'

##### SUMMARY
I'm seeing this in test with pfsensible/core:
```
____________________ TestPFSenseCertModule.test_cert_create ____________________
[gw1] linux -- Python 3.10.18 /opt/hostedtoolcache/Python/3.10.18/x64/bin/python

args = {'_ansible_keep_remote_files': False, '_ansible_remote_tmp': '/tmp', 'ca': 'testdel', 'name': 'cert1', ...}

    @_contextlib.contextmanager
    def set_module_args(args):
        """
        Context manager that sets module arguments for AnsibleModule
        """
        if '_ansible_remote_tmp' not in args:
            args['_ansible_remote_tmp'] = '/tmp'
        if '_ansible_keep_remote_files' not in args:
            args['_ansible_keep_remote_files'] = False
    
        try:
>           from ansible.module_utils.testing import patch_module_args
E           ModuleNotFoundError: No module named 'ansible.module_utils.testing'

../../community/internal_test_tools/tests/unit/plugins/modules/utils.py:35: ModuleNotFoundError

During handling of the above exception, another exception occurred:

self = <ansible_collections.pfsensible.core.tests.unit.plugins.modules.test_pfsense_cert.TestPFSenseCertModule testMethod=test_cert_create>

    def test_cert_create(self):
        """ test creation of a new cert """
        obj = dict(name='cert1', ca='testdel')
>       self.do_module_test(obj, command="create cert 'cert1'")

tests/unit/plugins/modules/test_pfsense_cert.py:80: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unit/plugins/modules/pfsense_module.py:188: in do_module_test
    self.assertIsNotNone(target_elt)
E   AssertionError: unexpectedly None
```
https://github.com/pfsensible/core/actions/runs/17030528174/job/48272652229

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below, use your best guess if unsure -->
/tests/unit/plugins/modules/utils.py
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
2.14.18
2.17.13
```

##### COLLECTION VERSION
<!--- Paste verbatim output from "ansible-galaxy collection list <namespace>.<collection>"  between the quotes
for example: ansible-galaxy collection list community.general
-->
```paste below
0.18.0
```

##### CONFIGURATION
<!--- Paste verbatim output from "ansible-config dump --only-changed" between quotes -->
```paste below

```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. target OS versions, network device firmware, etc. -->
GitHub actions

##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->

<!--- Paste example playbooks or commands between quotes below -->
```yaml

```

<!--- HINT: You can paste gist.github.com links for larger files -->

##### EXPECTED RESULTS
<!--- Describe what you expected to happen when running the steps above -->


##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->

<!--- Paste verbatim command output between quotes -->
```paste below

```

--- a/changelogs/fragments/160-set-module-args.yml
+++ b/changelogs/fragments/160-set-module-args.yml
@@ -1,0 +1,4 @@
+bugfixes:
+  - "Refactor ``ansible_collections.community.internal_test_tools.tests.unit.plugins.modules.utils.set_module_args()``
+     so that neither branch is executed during exception handling, to avoid backtraces in pytest when errors happen somewhere else
+     (https://github.com/ansible-collections/community.internal_test_tools/issues/159, https://github.com/ansible-collections/community.internal_test_tools/pull/160)."

--- a/tests/unit/plugins/modules/utils.py
+++ b/tests/unit/plugins/modules/utils.py
@@ -34,14 +34,17 @@ def set_module_args(args):
     try:
         from ansible.module_utils.testing import patch_module_args
     except ImportError:
-        # Before data tagging support was merged, this was the way to go:
-        serialized_args = to_bytes(json.dumps({'ANSIBLE_MODULE_ARGS': args}))
-        with patch.object(basic, '_ANSIBLE_ARGS', serialized_args):
-            yield
+        pass
     else:
         # With data tagging support, we have a new helper for this:
         with patch_module_args(args):
             yield
+        return
+
+    # Before data tagging support was merged, this was the way to go:
+    serialized_args = to_bytes(json.dumps({'ANSIBLE_MODULE_ARGS': args}))
+    with patch.object(basic, '_ANSIBLE_ARGS', serialized_args):
+        yield
 
 
 class AnsibleExitJson(Exception):

